erpnext/accounts/doctype/payment_entry/payment_entry.py

```diff
@@ -61,7 +63,23 @@ def on_cancel(self):
 		self.make_gl_entries(cancel=1)
 		self.update_advance_paid()
 		self.delink_advance_entry_references()
+
+	def validate_duplicate_entry(self):
+		reference_names = []
+		for d in self.get("references"):
+			print d.reference_name
+			if d.reference_name in reference_names:
+				frappe.throw(_("Duplicate entry in References {0} {1}").format(d.reference_doctype, d.reference_name))
+			reference_names.append(d.reference_name)
+	
 	
+	def validate_allocated_amount(self):
+		for d in self.get("references"):
+			if (flt(d.allocated_amount))> 0:
+				if d.allocated_amount > d.outstanding_amount:
+					frappe.throw(_("Allocated Amount cannot be greater than outstanding amount."))
```

Comment:
`frappe.throw(_("Row #{0}: Allocated Amount cannot be greater than outstanding amount.").format(d.idx))`